extensions/interactions/TextInput/directives/text-input-validation.service.ts

```diff
@@ -95,9 +95,53 @@ export class TextInputValidationService {
       stateName: string,
       customizationArgs: TextInputCustomizationArgs,
       answerGroups: AnswerGroup[], defaultOutcome: Outcome): Warning[] {
-    return this.getCustomizationArgsWarnings(customizationArgs).concat(
-      this.bivs.getAllOutcomeWarnings(
-        answerGroups, defaultOutcome, stateName));
+    var warningsList: Warning[] = [];
+
+    warningsList = warningsList.concat(
+      this.getCustomizationArgsWarnings(customizationArgs));
+
+    var seenStringsContains: string[] = [];
+    var seenStringsStartsWith: string[] = [];
+
+    for (var i = 0; i < answerGroups.length; i++) {
+      var rules = answerGroups[i].getRulesAsList();
+      for (var j = 0; j < rules.length; j++) {
+        var currentString = <string> rules[j].inputs.x;
+        if (rules[j].type === 'Contains') {
```

Comment:
Aren't there other rules as well that can cause issues ?

Reply:
I've now added validations related to the remaining two rules as well.